### PR TITLE
Do not silently ignore a store that has no cluster ID

### PR DESCRIPTION
**Pull request by kkaneda**
IIRC, once a store has successfully started, it should have a valid cluster ID and a node ID. If it doesn't have, something wrong should happen and we should return an error.

### CODE REVIEW

File: server/node.go [hunk 1]
```diff
@@ -234,17 +234,15 @@ func (n *Node) initStores(clock *hlc.Clock, engines []engine.Engine) error {
...
-			if s.Ident.StoreID == 0 {
-				return util.Error("cluster id set for node ident but missing store id")
-			}
-			capacity, err := s.Capacity()
-			if err != nil {
-				return err
-			}
-			log.Infof("initialized store %s: %+v", s, capacity)
-			n.lSender.AddStore(s)
+		if s.Ident.ClusterID == "" || s.Ident.NodeID == 0 {
```
Comment (petermattis) [inline server/node.go]: How would a node get into a state like this?

This deserves a small test.

**Comment by petermattis** [comment]
LGTM

File: server/node.go (hunk 1 above)
Comment (kkaneda) [inline server/node.go]: Sure, added a test. This shouldn't happen unless StoreIdent gets corrupted for some reason.

File: server/node_test.go [hunk 2]
```diff
@@ -42,7 +42,7 @@ import (
 // of engines. The server and node are returned. If gossipBS is not
 // nil, the gossip bootstrap address is set to gossipBS.
 func createTestNode(addr net.Addr, engines []engine.Engine, gossipBS net.Addr, t *testing.T) (
-	*rpc.Server, *Node) {
+	*rpc.Server, *Node, error) {
```
Comment (spencerkimball) [inline server/node_test.go]: I dislike this change as it just adds a lot of extra lines of test code without adding useful functionality. If something is an unrecoverable and fully-unexpected error in a test, we should just t.Fatal(err) as early as possible. The previous code was much simpler.

**Comment by spencerkimball** [comment]
Aside from my recommendation that we revert some of the cosmetic changes to createTestNode, LGTM

**Comment by kkaneda** [comment]
Sure, I'll revert the change. I made the change because the new test checks if `node.start` returns an error and avoiding ```t.Fatal``` was needed in such case.

**Comment by spencerkimball** [comment]
Ah, well then maybe you just have the new test break out the steps in createTestNode() necessary to reproduce the expected error so you can check without a log.Fatal happening.

**Comment by kkaneda** [comment]
Sure! Sent a PR to fix this: https://github.com/cockroachdb/cockroach/pull/519

Could you take a look? Thanks!

**Pull request merged**